### PR TITLE
Refactor FXIOS-7931 [v122] Window manager unit tests & more early refactors

### PR DESCRIPTION
**Pull request by mattreaganmozilla**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7931)

## :bulb: Description

Additional early prep work for iPad multi-window. Ongoing WIP. This PR:
- Updates to new `WindowManager` to lay foundation for coordinating state across multiple app windows
- Adds some initial unit tests

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

File: Client/Application/WindowManager.swift [hunk 1]
```diff
@@ -18,24 +25,64 @@ struct AppWindowInfo {
...
+    func windowDidClose(uuid: WindowUUID) {
+        updateWindow(nil, for: uuid)
+    }
+
     // MARK: - Internal Utilities
 
-    private func windowInfo(for windowUUID: WindowUUID, createIfNeeded: Bool = false) -> AppWindowInfo? {
+    private func updateWindow(_ info: AppWindowInfo?, for uuid: WindowUUID) {
+        assert(info != nil || windows.count > 1,
+               "Cannot remove the only active window in the app. This is a client error.")
```
Comment (OrlaM) [inline Client/Application/WindowManager.swift:55]: This looks like something that should send an error to Sentry, can you add a fatal error to the logger here?

**Review by OrlaM — APPROVED** [review]
Small nit on logging which you can feel free to update in a future PR, but otherwise LGTM.

File: Client/Application/WindowManager.swift (hunk 1 above)
Comment (mattreaganmozilla) [reply Client/Application/WindowManager.swift]: Updated to add Sentry logs; I thought this area of the app architecture might warrant a new logging category so I added `.window` to `LoggerCategory`, but I'm happy to use a different category if you prefer

Comment (mattreaganmozilla) [reply Client/Application/WindowManager.swift]: @OrlaM I just noticed this comment in `LoggerCategory`:

> `/// Do not add new categories unless discussing with the team beforehand.`

Will make sure I double-check on this before merging.

**Review by yoanarios — APPROVED** [review]

**Pull request merged**